server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -443,6 +447,16 @@ public CollectionStatistics collectionStatistics(String field) throws IOExceptio
         return collectionStatistics;
     }
 
+    /**
+     * Compute the leaf slices that will be used by concurrent segment search to spread work across threads
+     * @param leaves all the segments
+     * @return leafSlice group to be executed by different threads
+     */
+    @Override
+    public LeafSlice[] slices(List<LeafReaderContext> leaves) {
```

Comment:
I think no need to change method visibility here:

```suggestion
    protected LeafSlice[] slices(List<LeafReaderContext> leaves) {
```